pkg/snapshot/snapshotter/snapshotter.go

```diff
@@ -282,6 +286,15 @@ func (ssr *Snapshotter) takeFullSnapshot(isFinal bool) (*brtypes.Snapshot, error
 	// close previous watch and client.
 	ssr.closeEtcdClient()
 
+	var err error
+
+	// Update the snapstore object before taking every full snapshot
+	// Refer: https://github.com/gardener/etcd-backup-restore/issues/422
+	ssr.store, err = snapstore.GetSnapstore(ssr.snapstoreConfig)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create snapstore from configured storage provider: %v", err)
+	}
```

Comment:
Hm, not having a hash mechanism here will be a problem if we have any error when creating the snapshot. Then we'll create new Snapstores again and again.

Reply:
Didn’t get you exactly .... I didn’t add the hash mechanism here because it only runs once in 24 hour.

Reply:
When `takeFullSnapshot` errors then we immediately [continue](https://github.com/gardener/etcd-backup-restore/blob/71fafbb5bd1a6aea6e234e082a8237ab16d7a896/pkg/server/backuprestoreserver.go#L297) and try again, right? And in every retry we re-call `snapstore.GetSnapstore(ssr.snapstoreConfig)`.

Reply:
no, the link you provide is when `ownercheck` fails and it takes final full snapshot

Reply:
Okay, thanks.